activerecord/lib/active_record/associations/association_scope.rb

```diff
@@ -159,6 +159,14 @@ def add_constraints(scope, owner, refl, chain_head, chain_tail)
           scope
         end
 
+        def apply_scope(scope, table, key, value)
+          if scope.table == table
+            scope.where!(key => value)
```

Comment:
I had to go up 4 methods to confirm it was okay for this to mutate. I really just don't see the benefit of proliferating `where!`.